docs/managing-compute-resources.asciidoc

```diff
@@ -0,0 +1,117 @@
+[id="{p}-managing-compute-resources"]
+== Managing compute resources
+
+When a Pod is created it may request CPU and RAM resources. It may also specify the maximum resources that the containers are allowed to consume. Both Pod `limits` and `requests` can be set in the specification of any object managed by the operator (Elasticsearch, Kibana or the APM server). For more information about how this is used by Kubernetes please see https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/[Managing Compute Resources for Containers].
+
+[float]
+[id="{p}-custom-resources"]
+=== Set custom resources
+
+The `resources` can be customized in the `podTemplate` of an object.
+
+Here is an example for Elasticsearch:
+
+[source,yaml]
+----
+spec:
+  nodes:
+  - podTemplate:
+      spec:
+        containers:
+        - name: elasticsearch
+          env:
+          - name: ES_JAVA_OPTS
+            value: -Xms2048M -Xmx2048M
+          resources:
+            requests:
+              memory: 2Gi
+              cpu: 1
+            limits:
+              memory: 4Gi
+              cpu: 2
+----
+
+This example also demonstrates how to set the JVM memory options accordingly using the `ES_JAVA_OPTS` environment variable.
+
+The same applies for every object managed by the operator, here is how to set some custom resources for Kibana:
+
+[source,yaml]
+----
+spec:
+  podTemplate:
+    spec:
+      containers:
+      - name: kibana
+        resources:
+          requests:
+            memory: 1Gi
+            cpu: 1
```

Comment:
Nit: stay consistent in the mem/cpu declaration order to make reading more comfortable. k8s doc put always memory first.